test/tests/VehicleTest.cpp

```diff
@@ -0,0 +1,1218 @@
+}
+
+TileElement* get_station_platform(const CoordsXYRangedZ& coords)
+{
+    // TODO(janisozaur): do something here
+    return {};
+}
+
+void decrement_guests_in_park()
+{
+    // Probably overkill
+    assert(false);
+}
+
+void peep_sprite_remove(Peep* peep)
+{
+    // Probably overkill
+    assert(false);
+}
+
+void Ride::FormatStatusTo(Formatter& ft) const
+{
+    // Probably overkill
+    assert(false);
+}
+
+void viewports_invalidate(int32_t left, int32_t top, int32_t right, int32_t bottom, int32_t maxZoom)
+{
+    // Safe to ignore
+}
+
+bool Duck::IsFlying()
+{
+    // Probably overkill
+    assert(false);
+    return false;
+}
+
+bool Peep::SetName(std::string_view value)
+{
+    // Probably overkill
+    assert(false);
+    return true;
+}
+
+template<> bool SpriteBase::Is<Peep>() const
+{
+    // Probably overkill
+    assert(false);
+    return sprite_identifier == SpriteIdentifier::Peep;
+}
+
+template<> bool SpriteBase::Is<Guest>() const
+{
+    // Probably overkill
+    assert(false);
+    auto peep = As<Peep>();
+    return peep && peep->AssignedPeepType == PeepType::Guest;
+}
+
+template<> bool SpriteBase::Is<VehicleCrashParticle>() const
+{
+    // Probably overkill
+    assert(false);
+    auto* misc = As<MiscEntity>();
+    return misc && misc->SubType == MiscEntityType::CrashedVehicleParticle;
+}
+
+template<> bool SpriteBase::Is<CrashSplashParticle>() const
+{
+    // Probably overkill
+    assert(false);
+    auto* misc = As<MiscEntity>();
+    return misc && misc->SubType == MiscEntityType::CrashSplash;
+}
+
+template<> bool SpriteBase::Is<JumpingFountain>() const
+{
+    // Probably overkill
+    assert(false);
+    auto* misc = As<MiscEntity>();
+    return misc
+        && (misc->SubType == MiscEntityType::JumpingFountainSnow || misc->SubType == MiscEntityType::JumpingFountainWater);
+}
+
+template<> bool SpriteBase::Is<Balloon>() const
+{
+    // Probably overkill
+    assert(false);
+    auto* misc = As<MiscEntity>();
+    return misc && misc->SubType == MiscEntityType::Balloon;
+}
+
+template<> bool SpriteBase::Is<Duck>() const
+{
+    // Probably overkill
+    assert(false);
+    auto* misc = As<MiscEntity>();
+    return misc && misc->SubType == MiscEntityType::Duck;
+}
+
+template<> bool SpriteBase::Is<MoneyEffect>() const
+{
+    // Probably overkill
+    assert(false);
+    auto* misc = As<MiscEntity>();
+    return misc && misc->SubType == MiscEntityType::MoneyEffect;
+}
+
+Guest* Peep::AsGuest()
+{
+    // Probably overkill
+    assert(false);
+    return nullptr;
+}
+
+int32_t Guest::GetEasterEggNameId() const
+{
+    // TODO(janisozaur): do something here
+    return -1;
+}
+
+TEST(Vehicle, Create)
+{
+    [[maybe_unused]] Vehicle v{};
+}
+
+TEST(Vehicle, SimpleUpdate)
+{
+    Vehicle v{};
+    v.Update();
+}
+
+TEST(Vehicle, CableLiftUpdate)
+{
+    reset_sprite_list();
+    Ride& r = _rideList[0];
+    r = Ride{};
+    r.id = 0;
+    Vehicle* v = cable_lift_segment_create(r, 0, 0, 0, 0, 0, 0, true);
```

Comment:
Semi-random cable lift test, as it required pulling in a new file into the target